src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/OrganizationUserUserDetailsQuery.cs

```diff
@@ -37,4 +50,45 @@ public async Task<IEnumerable<OrganizationUserUserDetails>> GetOrganizationUserU
                 return o;
             });
     }
+
+    /// <summary>
+    /// Get the organization user user details, two factor enabled status, and
+    /// claimed status for the provided request.
+    /// </summary>
+    /// <param name="request">Request details for the query</param>
+    /// <returns>List of OrganizationUserUserDetails</returns>
+    public async Task<IEnumerable<(OrganizationUserUserDetails, bool, bool)>> Get(OrganizationUserUserDetailsQueryRequest request)
+    {
+        var organizationUsers = await GetOrganizationUserUserDetails(request);
+
+        var organizationUsersTwoFactorEnabled = await _twoFactorIsEnabledQuery.TwoFactorIsEnabledAsync(organizationUsers);
+        var organizationUsersClaimedStatus = await _getOrganizationUsersClaimedStatusQuery.GetUsersOrganizationClaimedStatusAsync(request.OrganizationId, organizationUsers.Select(o => o.Id));
+        var responses = organizationUsers
+            .Select(o =>
+            {
+                var userTwoFactorEnabled = organizationUsersTwoFactorEnabled.FirstOrDefault(u => u.user.Id == o.Id).twoFactorIsEnabled;
+                var claimedByOrganization = organizationUsersClaimedStatus[o.Id];
+                var orgUser = (o, userTwoFactorEnabled, claimedByOrganization);
+
+                return orgUser;
+            });
+
+        return responses;
+    }
+
+    /// <summary>
+    /// Get the organization users user details, two factor enabled status, and
+    /// claimed status for confirmed users.
+    /// </summary>
+    /// <param name="request">Request details for the query</param>
+    /// <returns>List of OrganizationUserUserDetails</returns>
+    public async Task<IEnumerable<(OrganizationUserUserDetails, bool, bool)>> GetConfirmed(OrganizationUserUserDetailsQueryRequest request)
+    {
+
+        var result = await Get(request);
+        var responses = result.Where(r => r.Item1.Status.Equals(OrganizationUserStatusType.Confirmed));
+
+        return responses;
+    }
```

Comment:
I like how this is just a wrapper around another public method, that is generally the approach I prefer. However, it means we're getting a lot of data (2FA and claimed status) for users that we don't need. So in this case I do actually recommend some light code duplication so that we get the orgUsers, filter for the ones we want, and _then_ get the 2FA and confirmed status just for those users.

You could use a private method, but the methods in this class are so short and the heavy lifting is already encapsulated in the other queries, so I think a copy/paste & tweak of `Get` is fine.